make/ksctl.mk

```diff
@@ -1,7 +1,62 @@
-KSCTL_GH_OWNER=kubesaw
-KSCTL_GH_BRANCH=master
+USE_INSTALLED_KSCTL = false
 
 BIN_DIR := $(shell pwd)/build/_output/bin
 
+ifneq ($(USE_INSTALLED_KSCTL),true)
+KSCTL_BIN_DIR = ${BIN_DIR}/
+endif
+
+.PHONY: ksctl
 ksctl:
-	GOBIN=${BIN_DIR} CGO_ENABLED=0 go install github.com/${KSCTL_GH_OWNER}/ksctl/cmd/ksctl@${KSCTL_GH_BRANCH}
+ifneq ($(USE_INSTALLED_KSCTL),true)
+	@$(MAKE) get-ksctl-and-install --no-print-directory
+else
+	@echo "Using local version of ksctl"
+endif
+
+get-ksctl-and-install:
+ifeq ($(strip $(KSCTL_REPO_PATH)),)
+    ifneq ($(CI),)
+        ifneq ($(GITHUB_ACTIONS),)
+			$(eval BRANCH_NAME = ${GITHUB_HEAD_REF})
+			$(eval AUTHOR_LINK = https://github.com/${AUTHOR})
+        else
+			$(eval AUTHOR_LINK = $(shell jq -r '.refs[0].pulls[0].author_link' <<< $${CLONEREFS_OPTIONS} | tr -d '[:space:]'))
+			@echo "found author link ${AUTHOR_LINK}"
+			$(eval BRANCH_NAME := $(shell jq -r '.refs[0].pulls[0].head_ref' <<< $${CLONEREFS_OPTIONS} | tr -d '[:space:]'))
+        endif
+		@echo "using author link ${AUTHOR_LINK}"
+		@echo "detected branch ${BRANCH_NAME}"
+		# check if a branch with the same ref exists in the user's fork of ksctl repo
+		$(eval REMOTE_KSCTL_BRANCH := $(shell curl ${AUTHOR_LINK}/ksctl.git/info/refs?service=git-upload-pack --output - 2>/dev/null | grep -a "refs/heads/${BRANCH_NAME}$$" | awk '{print $$2}'))
+		@echo "branch ref of the user's fork: \"${REMOTE_KSCTL_BRANCH}\" - if empty then not found"
+		# check if the branch with the same name exists, if so then merge it with master and use the merge branch, if not then use master
+        ifneq ($(REMOTE_KSCTL_BRANCH),"")
+			# define temp dir
+			$(eval KSCTL_REPO_PATH := /tmp/ksctl)
+			# delete to have clear environment
+			rm -rf ${KSCTL_REPO_PATH}
+
+			git config --global user.email "devtools@redhat.com"
+			git config --global user.name "Devtools"
+			# clone
+			git clone --depth=1 https://github.com/kubesaw/ksctl.git ${KSCTL_REPO_PATH}
+			# add the user's fork as remote repo
+			git --git-dir=${KSCTL_REPO_PATH}/.git --work-tree=${KSCTL_REPO_PATH} remote add external ${AUTHOR_LINK}/ksctl.git
+			# fetch the branch
+			git --git-dir=${KSCTL_REPO_PATH}/.git --work-tree=${KSCTL_REPO_PATH} fetch external ${REMOTE_KSCTL_BRANCH}
+			# merge the branch with master
+			git --git-dir=${KSCTL_REPO_PATH}/.git --work-tree=${KSCTL_REPO_PATH} merge --allow-unrelated-histories --no-commit FETCH_HEAD
```

Comment:
just curious - why do we need to clone both the upstream and the fork and merge them ? was there any problem with cloning what was specified in the `KSCTL_GH_OWNER` variable and `KSCTL_GH_BRANCH` and building and installing that ?

Reply:
also could there be merge conflicts when doing this ?

Reply:
Ok now I see , I think we want it to work in the same way as the pairing for the other repos, so if there is a conflict the author must merge/fix and then rerun the tests. 

Not sure why - but I was thinking that initially we would have a simpler mechanism that just clones a given ksctl repo , builds and installs it.

Reply:
we want to test the final state - when the PR is merged to master. We don't want test only the changes in the PR (without combining them with master) as the PR may be outdated and it doesn't have to contain the latest changes important for the functions of the commands.